### ISSUE TITLE
[BUG] - Provisioning - No Alerts displayed in the Alert Table at start up.

### ISSUE THREAD

**Issue by Filipe777** [grafana/grafana#13319]
### What Grafana version are you using?
5.2.4

### What datasource are you using?
Elasticsearch 6.4.0

### What OS are you running grafana on?
MacOS High Sierra

### What did you do?
Provisioned 4 dashboards made specifically for Alerting in a separate folder.

dashboards.yaml: 
![image](https://user-images.githubusercontent.com/31474068/45681402-5778de80-bb35-11e8-8bda-48c08ce8ec70.png)

### What was the expected result?
Start up grafana with the 4 dashboards in place and verify that the Alert Tables are displaying alerts status.

![image](https://user-images.githubusercontent.com/31474068/45681540-a9b9ff80-bb35-11e8-9f6e-640c021a9f7e.png)

Table config:
![image](https://user-images.githubusercontent.com/31474068/45681672-074e4c00-bb36-11e8-9dec-2d441503a08a.png)

### What happened instead?
**No alerts** displayed in the alerts tables.
![image](https://user-images.githubusercontent.com/31474068/45681615-db32cb00-bb35-11e8-9faf-a3821fce0034.png)
  
##### To have the alerts being displayed I have to go into the alert table config, change the **Folder** from **General** to **Alerts** and back to **General**

![image](https://user-images.githubusercontent.com/31474068/45681851-7deb4980-bb36-11e8-85f2-60df3d1a201b.png)

![image](https://user-images.githubusercontent.com/31474068/45681880-90658300-bb36-11e8-8fa2-b7972e44ec2d.png)

![image](https://user-images.githubusercontent.com/31474068/45681938-b5f28c80-bb36-11e8-8d36-edfcb9f304bd.png)

Also, I wonder why the alert tables have the **Folder** defined as **General** when the dashboard was made with **Alerts** as **Folder**. The only reference to a folder in the dashboard.json file is the `"folderId"` which I don't know what `folderId` is my **Alerts** folder.

![image](https://user-images.githubusercontent.com/31474068/45682162-74aeac80-bb37-11e8-996c-e60116203849.png)

 Thank you!

**Comment by torkelo** [comment]
your provisioning folders should not be nested (so don't provisioning a folder who's parent is already provisioned by another rule)

**Comment by marefr** [comment]
To be able to filter alert list panel by folder you need to specify an existing `folderId`.

**Comment by Filipe777** [comment]
Thank you for your fast reply! I really appreciate it.

Ok, both issues (also #13316 ) are then related to the same folder nesting approach. 

@torkelo , I've tried to have specific folders for the dashboards, a set of dashboards in `/var/lib/grafana/dashboards/general` and the Alerts dashboards in `/var/lib/grafana/dashboards/alerts` but I have the same results. I've also tried putting the **Alerts** dashboards folder in `/var/lib/grafana/alerts` but they also appeared in `general` folder. 

@marefr , how can I get my **Alerts** `folderId`? So I can specify it in the alerts tables. It's showing as `12` in the json file but I didn't specified it.
